sources/packages/backend/apps/api/src/services/student-file/student-file.service.ts

```diff
@@ -37,7 +52,39 @@ export class StudentFileService extends RecordDataModelService<StudentFile> {
     newFile.fileContent = createFile.fileContent;
     newFile.student = { id: studentId } as Student;
     newFile.creator = { id: auditUserId } as User;
-    return this.repo.save(newFile);
+
+    const summary = new ProcessSummary();
+    let savedFile: StudentFile;
+    // Save the file to db.
+    try {
```

Comment:
have a single try and remove the let for savedFile and change it to const. Something like below.

`  try {
    const savedFile = await this.repo.save(newFile);

    summary.info(`Adding virus scan for the file: ${savedFile.fileName}.`);
    await this.virusScanQueue.add({
      uniqueFileName: savedFile.uniqueFileName,
      fileName: savedFile.fileName,
    });
    await this.repo.update(
      { uniqueFileName: savedFile.uniqueFileName },
      { virusScanStatus: VirusScanStatus.InProgress },
    );
    summary.info(
      `File ${savedFile.fileName} has been added to the virus scan queue.`,
    );

    return savedFile;
  } catch (error) {
    summary.error(
      `Error while saving or enqueueing the file ${newFile.fileName} for virus scanning.`,
      error,
    );
    this.logger.error(`Error saving the file: ${error}`);
    throw error;
  } finally {
    this.logger.logProcessSummary(summary);
  } 
  `

Reply:
I believe we need two different `try` blocks.